File: eazy/链表/ListNode.java
package 链表;

/**
 * @ClassName 链表.ListNode
 * @Description 链表公共类
 * @Author MatthewHan
 * @Date 2020/4/30 10:28
 * @Version 1.0
 **/
public class ListNode {
    public int val;
    public ListNode next;

    public ListNode(int x) {
        val = x;
    }

    @Override
    public String toString() {
        return "ListNode{" +
                "val=" + val +
                ", next=" + next +
                '}';
    }
}

File: lang-py/ListNode.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

    def __str__(self):
        ans = str(self.val) + ' --> '
        while self.next:
            ans += str(self.next.val)
            ans += " --> "
            self.next = self.next.next
        return ans
